2-collectors/scala-stream-collector/src/main/scala/com.snowplowanalytics.snowplow.collectors/scalastream/CollectorService.scala

```diff
@@ -54,43 +59,122 @@ class CollectorServiceActor(collectorConfig: CollectorConfig,
 }
 
 /**
- * Companion object for the CollectorService class
- */
+  *
+  * Companion object for the CollectorService class
+  */
 object CollectorService {
   private val QuerystringExtractor = "^[^?]*\\?([^#]*)(?:#.*)?$".r
 }
 
 // Store the route in CollectorService to be accessed from
 // both CollectorServiceActor and from the testing framework.
 class CollectorService(
-    responseHandler: ResponseHandler,
-    context: ActorRefFactory) extends HttpService {
+                        collectorConfig: CollectorConfig,
+                        responseHandler: ResponseHandler,
+                        context: ActorRefFactory) extends HttpService {
   def actorRefFactory = context
 
+  val cookieName = collectorConfig.cookieName
+
   // TODO: reduce code duplication here
   val collectorRoute = {
     post {
       path(Segment / Segment) { (path1, path2) =>
-        optionalCookie("sp") { reqCookie =>
+        cookieIfWanted(cookieName) { reqCookie =>
```

Comment:
I didn't change this, this comes from the master mainline SP sync so I think we are stuck with it unless we want to diverge further from the mainline